src/ManagedArray.inl

```diff
@@ -110,6 +110,8 @@ CHAI_HOST_DEVICE ManagedArray<T>::ManagedArray(ManagedArray const& other):
 
   m_active_pointer = static_cast<T*>(m_resource_manager->move(const_cast<T_non_const*>(m_active_pointer), m_pointer_record));
 
+  migrateInnerData();
```

Comment:
Can we rename this, something like 'moveInnerData` or `copyInnerData`. "MIgrate" means we are mixing verbs.

Reply:
Changed to "move".